assessments/projects/openkruise/self-assessment.md

```diff
@@ -0,0 +1,563 @@
+</figcaption>
+</figure>
+
+#### Actors
+
+- `kruise-operator`
+  - `kubectl-kruise`
+  - Golang and Java clients
+- `kruise-manager`
+  - `kruise-controller`
+  - `kruise-webhook`
+- `kruise-daemon`
+- `kube-apiserver`
+- `kubelet`
+- Pods and containers
+
+##### `kruise-operator`
+
+The operator is the person administrating or operating the system. This person
+has authorization to access the OpenKruise system and perform actions. The
+operator interacts with OpenKruise using the CLI or client SDK.
+
+##### `krusie-manager`
+
+This is a control plane component that runs controllers and webhooks.
+Logically, each controller is a separate process but to reduce complexity they
+are compiled into the same binary running on the API server. It consists of a
+`kruise-controller` and `kruise-webhook`.
+
+- `kruise-controller` - Responsible for checking the that the configuration is
+  in the desiredstate based on the user-configuration. It checks the resources
+  across all of the nodes and ensures they are up-to-date.
+
+- `kruise-webhook` - Used for admission control. It intercepts，validates and
+  potentially mutates requests coming from the `kruise-operator`. The
+  `kruise-webhook` is important since the `kube-apiserver` will fail the
+  request if the calling to the `kruise-webhook` fails.
+
+##### `kruise-daemon`
+
+This runs on every node and manages things like image pre-download and
+container restarting. It interacts with the `kruise-manager` indirectly by
+calling the API Server which calls through.
+
+##### `kube-apiserver`
+
+The Kubernetes API server validates and configures data for the api objects.
+This include pods, services, replicationcontrollers, and other things. It
+provides the frontend to the clusters shared state.
+
+##### `kubelet`
+
+The `kubelet` runs on each node. It's the primary "node-agent" that can register
+the node with the apiserver so that pods on that node can be managed.
+
+#### Actions
+
+These are the steps that a project performs in order to provide some service
+or functionality. These steps are performed by different actors in the system.
+Note, that an action need not be overly descriptive at the function call level.
+It is sufficient to focus on the security checks performed, use of sensitive
+data, and interactions between actors to perform an action.
+
+For example, the access server receives the client request, checks the format,
+validates that the request corresponds to a file the client is authorized to
+access, and then returns a token to the client. The client then transmits that
+token to the file server, which, after confirming its validity, returns the file.
+
+##### Operator based action
+
+One of the possible ways of interacting with the system are if a
+`kruise-operator` takes a manual action to administer some part of the system
+by using the CLI or a client SDK.
+
+###### Actors
+
+- `kruise-operator` - The person administrating or operating the system
```

Comment:
Is this a term used by the Kruise team? I'm quite surprised if that's the case, because [Kubernetes Operators](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) are already a well established thing. Instead of saying "operator" to describe the person using the system, consider simply saying "user."

Reply:
This term was originally introduced by us. I've changed `krusie-operator` to user where we've needed to reference it in the document to make sure there is no confusion.